### PR TITLE
[SAMPLES][GNA] Update C++ speech sample with new config API

### PR DESCRIPTION
**Pull request by nkogteva**
### Details:
 - *Use new configuration properties in C++ speech sample*

### Tickets:
 - *78938*

### CODE REVIEW

File: samples/cpp/speech_sample/utils.hpp [hunk 1]
```diff
@@ -361,30 +361,103 @@ void sum_performance_counters(std::map<std::string, ov::ProfilingInfo> const& pe
...
+    };
+    std::map<std::string, float> result;
+    auto scale_factor_strings = split(values_string, ',');
+    for (auto& scale_factor_string : scale_factor_strings) {
+        auto values = split(scale_factor_string, ':');
+        if (values.size() == 1) {
+            auto scale_factor = std::stof(values.at(0));
+            check_sf(scale_factor);
+            for (auto& input : inputs) {
+                result[input.get_any_name()] = scale_factor;
```
Comment (mryzhov) [inline samples/cpp/speech_sample/utils.hpp:448]: Seems like the resulting map will be overwritten by the last scale_factor.

Comment (nkogteva) [reply samples/cpp/speech_sample/utils.hpp:453]: @mryzhov Additional check was added

File: samples/cpp/speech_sample/utils.hpp [hunk 2]
```diff
@@ -361,30 +361,108 @@ void sum_performance_counters(std::map<std::string, ov::ProfilingInfo> const& pe
...
+    }
+    if (!count) {
+        std::stringstream ss;
+        ss << "Incorrect node name '" + node_name << "'! ";
+        ss << "Try one of the following names: [ ";
+        for (auto&& name : any_names) {
+            ss << name << " ";
+        }
+        ss << "]";
+        throw std::logic_error(ss.str());
```
Comment (mgrdv) [inline samples/cpp/speech_sample/utils.hpp:421]: Why do we throw an error here after adding data to the stream?

File: samples/cpp/speech_sample/utils.hpp [hunk 3]
```diff
@@ -361,30 +361,108 @@ void sum_performance_counters(std::map<std::string, ov::ProfilingInfo> const& pe
...
+            }
+            auto scale_factor = std::stof(values.at(0));
+            check_sf(scale_factor);
+            for (auto& input : inputs) {
+                result[input.get_any_name()] = scale_factor;
             }
-            scaleFactorInput.push_back(outStr);
-            i++;
+        } else {
+            auto scale_factor = std::stof(values.back());
```
Comment (mgrdv) [inline samples/cpp/speech_sample/utils.hpp:456]: what if  size of values is empty after splitting?

File: samples/cpp/speech_sample/utils.hpp (hunk 2 above)
Comment (nkogteva) [reply samples/cpp/speech_sample/utils.hpp:421]: @mgrdv I don't understand the question. Error message is added to the stream.

**Review by mgrdv — APPROVED** [review]

**Review by antonrom23 — APPROVED** [review]

**Comment by antonrom23** [comment]
@nkogteva , build errors on windows:
openvino\samples\cpp\speech_sample\utils.hpp(438): error C2220: the following warning is treated as an error
openvino\samples\cpp\speech_sample\utils.hpp(438): warning C4101: 'ex': unreferenced local variable
Please, fix it.

**Pull request merged**